packages/odata2ts/src/data-model/DataModelDigestion.ts

```diff
@@ -23,6 +23,8 @@ export abstract class Digester<S extends Schema<ET, CT>, ET extends EntityType,
 
   private model2Type: Map<string, DataTypes> = new Map<string, DataTypes>();
 
+  private visitedModelsCircuitBreaker: string[] = [];
```

Comment:
instead of using a private property / state here. Can you pass  this array directly into the function, e.g. `collectBaseClassPropsAndKeys(model, visitedModels)` ?